feat & fix: better handling about the access rate

zopanix opened the pull request:
This contribution contains two changes:

* fix: which changes the unit of the access fee from $/kwh to $/day
* feat: which creates a entity that includes the daily access fee for a more accurate cost calculation

christophe-scalepad committed:
fix: Change access cost unit to $ per day

As explained here: https://www.hydroquebec.com/residentiel/espace-clients/tarifs/option-credit-hivernal-tarification.html

christophe-scalepad committed:
feat: Add new entity called `hilo_rate_current_total`

This new rate also includes the access rate which provides you with a
more accurate total cost

ic-dev21 committed:
Update sensor.py

Retiré un update de trop, lint check

ic-dev21 committed:
Update sensor.py

Ajout d'un display precision dans le GUI, j'avais à peu près 17 décimales

ic-dev21 committed:
Merge branch 'main' into pr/793

ic-dev21 committed:
Merge branch 'main' into pr/793

ic-dev21 commented on custom_components/hilo/sensor.py:
```diff
@@ -201,10 +201,23 @@ def create_energy_entity(hilo, device):
+        hilo._hass,
+        ["sensor.hilo_rate_current"],
+        hilo_rate_current_total._handle_state_change,
+    )
```
Pas certain de comprendre à quoi ça sert, hilo_rate_current s'occupe du taux actuel (première tranche de 40kwh, vs le reste).

ic-dev21 committed:
Make properly async and linting

ic-dev21 committed:
Further lazy formatting

ic-dev21 commented:
@zopanix j'ai pas eu de réponse de ta part mais je te laisse regarder les changements proposés.

ic-dev21 committed:
Merge branch 'main' of https://github.com/zopanix/hilo into pr/793

ic-dev21 committed:
Merge branch 'main' into pr/793

ic-dev21 committed:
Fixing the math

Les unités de mesure du cost sensor ne fonctionnaient pas, ajusté pour que ça donne un résultat en dollars.

ic-dev21 committed:
Merge branch 'main' into main

ic-dev21 committed:
Merge branch 'main' into pr/793

ic-dev21 committed:
No need to pass period

Already passed in the energymeters used.
